src/elements/EditableTable.tsx

```diff
@@ -0,0 +1,262 @@
+    const footerRow: FooterRow = calculateFooterRow(cols, rows);
+
+    const handleMouseEnter = (event: React.MouseEvent, row: Row, column: string) => {
+        raiseEvent("custom", {
+            eventType: "mouseEnter",
+            rowData: { rows: data, row: row, column: column },
+        });
+        onMouseEnter?.(row);
+    };
+
+    const handleMouseLeave = (event: React.MouseEvent, row: Row, column: string) => {
+        raiseEvent("custom", {
+            eventType: "mouseLeave",
+            rowData: { rows: data, row: row, column: column },
+        });
+        onMouseLeave?.(row);
+    };
+
+    const handleCellChange = (
+        event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+        row: Row,
+        column: Column
+    ) => {
+        if (column?.parse) {
+            row[column.name] = column.parse(event, row, column);
+        } else if ((event.target as any).type === "number") {
+            row[column.name] = Number(event.target.value);
+        } else {
+            row[column.name] = event.target.value;
+        }
+        if (rowCalculation) {
+            rowCalculation(data, row);
+        }
+        raiseEvent("custom", {
+            eventType: "cellChange",
+            rowData: { rows: data, row: row, column: column },
+        });
+        setData([...data]);
+        setValue(data);
+    };
+
+    return (
+        <Box maxHeight={maxHeight} maxWidth={maxWidth} sx={{ overflow: "auto" }}>
+            <Table
+                size={size}
+                stickyHeader={stickyHeader}
+                sx={{
+                    // Hack required to override .gcx-forms.defaults
+                    ["& span"]: {
+                        display: "inline-flex !important",
+                    },
+                }}
+            >
+                <TableHead>
+                    <TableRow>
+                        {includeFooter && <TableCell />}
+                        {cols.map((col) => (
+                            <TableCell align={col.align} key={col.name}>
+                                {col.alias ?? col.name}
+                            </TableCell>
+                        ))}
+                    </TableRow>
+                </TableHead>
+                <TableBody>
+                    {data.map((row, index) => {
+                        return (
+                            <TableRow hover key={index}>
+                                {includeFooter && <TableCell />}
+
+                                {cols.map((col, index) => {
+                                    const value = row[col.name];
+                                    const formattedValue = col.format ? col.format(value) : value;
+                                    return (
+                                        <TableCell
+                                            align={col.align}
+                                            key={index}
+                                            onMouseEnter={(event) =>
```

Comment:
Do we really want these events on the cell level? The regular table has them on the row.

Reply:
Sure. I'll move them.